packages/legend-query-builder/src/components/explorer/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -282,32 +292,70 @@ export const QueryBuilderPropertySearchPanel = observer(
     const { queryBuilderState, triggerElement } = props;
     const explorerState = queryBuilderState.explorerState;
     const propertySearchPanelState = explorerState.propertySearchState;
+    const hasHiddenResults =
+      propertySearchPanelState.isOverSearchLimit &&
+      propertySearchPanelState.filteredPropertyNodes.length !== 0;
     const searchInputRef = useRef<HTMLInputElement>(null);
-    const changePropertyName: React.ChangeEventHandler<HTMLInputElement> = (
-      event,
-    ) => {
+
+    const handleEnter = (): void => searchInputRef.current?.focus();
+
+    const debouncedSearch = useMemo(
+      () => debounce(() => propertySearchPanelState.search(), 100),
+      [propertySearchPanelState],
+    );
+    const onSearchPropertyTextChange: React.ChangeEventHandler<
+      HTMLInputElement
+    > = (event) => {
       propertySearchPanelState.setSearchText(event.target.value);
-      propertySearchPanelState.refreshPropertyState();
-      if (propertySearchPanelState.searchText.length >= 3) {
-        propertySearchPanelState.fetchMappedPropertyNodes(
-          propertySearchPanelState.searchText,
-        );
-      }
+      debouncedSearch();
     };
-    const clearPropertyName = (): void => {
+
+    const clearSearchPropertyName = (): void => {
       propertySearchPanelState.setSearchText('');
-      propertySearchPanelState.refreshPropertyState();
+      propertySearchPanelState.resetPropertyState();
     };
+
+    const handleSearchMode = (
+      event: React.ChangeEvent<HTMLInputElement>,
+    ): void => {
+      const searchMode = (event.target as HTMLInputElement)
+        .value as SEARCH_MODE;
+      propertySearchPanelState.changeModeOfSearch(searchMode);
+      propertySearchPanelState.resetPropertyState();
+      propertySearchPanelState.fetchMappedPropertyNodes(
+        propertySearchPanelState.searchText,
+      );
+    };
+
     const handleClose = (): void => {
-      clearPropertyName();
+      clearSearchPropertyName();
       propertySearchPanelState.setIsSearchPanelOpen(false);
     };
+
+    const getDocumentationToImplementSvp = (): void => {
+      queryBuilderState.applicationStore.assistantService.setIsOpen(true);
+      queryBuilderState.applicationStore.assistantService.setIsSearchPanelTipsOpen(
+        false,
+      );
+      queryBuilderState.applicationStore.assistantService.setSearchText(
```

Comment:
Can we be fancy here and implement a mechanism to open any entry in virtual assistant given the documentation key? This implementation is brittle as it's subjected to the content of a doc which is not part of this codebase :) This would be really nice!

Reply:
yes I was hoping for a method like that and made a note to implement and then proceeded to not do it :,) added 👍